### ISSUE TITLE
[@angular-eslint/template/*] rules are not run against inline templates

### ISSUE THREAD

**Issue by snebjorn** [angular-eslint/angular-eslint#347]
The template rules are not validating the inline templates, only the HTML files.

Given the below config (I stripped to as basic as possible)

```jsonc
// eslintrc.json
{
  "root": true,
  "ignorePatterns": ["projects/**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "parserOptions": {
        "project": ["tsconfig.json", "e2e/tsconfig.json"],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/ng-cli-compat",
        "plugin:@angular-eslint/ng-cli-compat--formatting-add-on",
        "plugin:@angular-eslint/template/process-inline-templates",
      ],
      "plugins": ["eslint-plugin-unicorn", "eslint-plugin-no-null"],
      "rules": {}
    },
    {
      "files": ["*.html"],
      // there's an issue with inline angular templates
      // see https://github.com/angular-eslint/angular-eslint/issues/105
      "excludedFiles": ["*inline-template-*.component.html"],
      "extends": ["plugin:@angular-eslint/template/recommended"],
      "rules": {
        "@angular-eslint/template/accessibility-alt-text": "error"
      }
    }
  ]
}
```

Then this inline template won't report any errors. It should complain about missing alt-text. The rule does work as errors are correctly reported from HTML files.
```ts
@Component({
  selector: "app-root",
  template: `
    <img src="asdf" />
  `,
})
export class AppComponent {}
```
Moving the rule into the TypeScript section doesn't work.
```
{
  "overrides": [
    {
      "files": ["*.ts"],
      "rules": {
        "@angular-eslint/template/accessibility-alt-text": "error"
      }
    }
  ]
}
```
As it'll throw this error
>  Error while loading rule '@angular-eslint/template/accessibility-alt-text': You have used a rule which requires '@angular-eslint/template-parser' to be used as the 'parser' in your ESLint config.

----
```
Angular CLI: 11.1.4
Node: 14.15.1
OS: win32 x64

Angular: 11.1.2
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.1
@angular-devkit/build-angular   0.1101.4
@angular-devkit/core            7.3.10
@angular-devkit/schematics      7.3.10
@angular/cli                    11.1.4
@schematics/angular             11.2.1
@schematics/update              0.1101.4
rxjs                            6.6.3
typescript                      4.1.5

@angular-eslint/eslint-plugin   1.2.0
@angular-eslint/eslint-plugin-template 1.2.0
@angular-eslint/schematics      1.2.0
@angular-eslint/template-parser 1.2.0
eslint                          7.20.0
```

**Comment by JamesHenry** [comment]
Hi @snebjorn, you are explicitly telling ESLint to ignore those files, so it is logical that it is ignoring those files. I'm really not sure what else to say here other than please don't ignore the files if you don't want them to be ignored

**Comment by snebjorn** [comment]
@JamesHenry I did try to put them in the *.ts section so they're not ignored. But that causes this to happen.

> Moving the rule into the TypeScript section doesn't work.
> 
> ```
> {
>   "overrides": [
>     {
>       "files": ["*.ts"],
>       "rules": {
>         "@angular-eslint/template/accessibility-alt-text": "error"
>       }
>     }
>   ]
> }
> ```
> 
> As it'll throw this error
> 
> > Error while loading rule '@angular-eslint/template/accessibility-alt-text': You have used a rule which requires '@angular-eslint/template-parser' to be used as the 'parser' in your ESLint config.

I'm not familiar enough with ESLint to know if adding the template-parser to the ts section is the correct thing to do.
I couldn't find any mention in the example configs about how to handle make sure that *.html templates and inline templates have the same set of rules applied.

**Comment by JamesHenry** [comment]
@snebjorn in your example config above, the only thing you need to do is remove the `excludedFiles` line, not move things around.

```diff

// eslintrc.json
{
  "root": true,
  "ignorePatterns": ["projects/**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "parserOptions": {
        "project": ["tsconfig.json", "e2e/tsconfig.json"],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/ng-cli-compat",
        "plugin:@angular-eslint/ng-cli-compat--formatting-add-on",
        "plugin:@angular-eslint/template/process-inline-templates",
      ],
      "plugins": ["eslint-plugin-unicorn", "eslint-plugin-no-null"],
      "rules": {}
    },
    {
      "files": ["*.html"],
-     // there's an issue with inline angular templates
-     // see https://github.com/angular-eslint/angular-eslint/issues/105
-     "excludedFiles": ["*inline-template-*.component.html"],
      "extends": ["plugin:@angular-eslint/template/recommended"],
      "rules": {
        "@angular-eslint/template/accessibility-alt-text": "error"
      }
    }
  ]
}
```

That line of config is what is ignoring the relevant files, and you just need to remove it to stop ignoring the files.

I am not sure why you are referencing that prettier issue in your config, because it does not apply to your setup based on what you have shared.

If you do end up using `eslint-plugin-prettier` at some point then this is the setup you need: https://github.com/angular-eslint/angular-eslint/issues/105#issuecomment-792271374 but as I say right now you aren't using it so there is no reason for you to be using `excludedFiles` anywhere in your config.

Hope that helps!
